src/lib/comms/include/sol-socket.h

```diff
@@ -0,0 +1,315 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <sol-network.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @defgroup Socket Socket
+ * @ingroup Comms
+ *
+ * @brief API to create an endpoint communication using sockets.
+ *
+ * @{
+ */
+
+struct sol_socket;
+
+struct sol_socket_default_options {
```

Comment:
ah, this `_default_` here makes no sense, call it "base" or just remove the specializer.